### ISSUE TITLE
User-provided section labels in markdown files are ignored by doxygen 1.9.7

### ISSUE THREAD

**Issue by diegodiasm** [doxygen/doxygen#10089]
Doxygen 1.9.7 ignores user-provided section labels in markdown files. In prior versions of doxygen (e.g. 1.9.1 and 1.9.4), if a file (e.g. `l4vio_readme.md`) is included in a doxygen project and starts with the following line:

`# Section name {#label}`

then `label` could be typed in a different file (e.g. `virtio-net-switch.dox`)  to link to the page written in the markdown file which defines this label (e.g. `l4vio_readme.md`).

In doxygen 1.9.7, writing `label` to refer to the page does not work. Instead, the markdown page is named according to the path to the file. In the [minimal working example](https://github.com/doxygen/doxygen/files/11564564/mwe-user-defined-markdown-labels.zip), which follows the structure below, this means that the markdown page will be named `md_pkg_2virtio-net-switch_2doc_2l4vio__readme.html`, and one would need to write `md_pkg_2virtio-net-switch_2doc_2l4vio__readme` to link to it.

```
.
├── Doxyfile
├── l4vio_readme.md
├── pkg
│   └── virtio-net-switch
│       └── doc
│           ├── l4vio_readme.md
│           └── virtio-net-switch.dox
└── server_overview.dox
```
(**Note**: The issue is _independent of the existence of sub-directories_. I've used sub-directories to illustrate that doxygen performs some path-based naming for the page.)

Is this a bug or an intended change of behaviour? If this is intended, what is the reason to ignore user-provided labels when naming pages?

**Comment by albert-github** [comment]
At first I thought the problem has recently been solved in the current master as part of the fix for @10056:
```
Commit: 13f5b6fee413d31404cf403a9855b7a9d257bc4f [13f5b6f]
Date: Tuesday, May 23, 2023 8:45:39 PM
issue #10056 [1.9.7] Markdown Header ID ignored in subdirectories
```
but I see that there is still the warning:
```
/pkg/virtio-net-switch/doc/virtio-net-switch.dox:11: warning: unable to resolve reference to 'label' for \ref command
```

The `#label` is defined in pkg/virtio-net-switch/doc/l4vio_readme.md 
```
# l4vio_switch, a virtual network switch   {#label}

The virtual network switch connects multiple clients with a virtual network
connection.

---- output -----
@page label l4vio_switch, a virtual network switch\ilinebr @ianchor{l4vio_switch, a virtual network switch} l4vio_switch-a-virtual-network-switch\ilinebr

The virtual network switch connects multiple clients with a virtual network
connection.
```

@doxygen looks like another case.

**Comment by doxygen** [comment]
@diegodiasm Please verify if the latest commit from master fixes the problem for you.
@albert-github Logic for the previous fix was wrong. Corrected now.

**Comment by diegodiasm** [comment]
@doxygen : Thanks for the fix. It solved the problem and works as expected. I am closing the issue. Thanks!

**Comment by lahwaacz** [comment]
@albert-github If you don't mind asking, when can we expect a release with this fix?

**Comment by albert-github** [comment]
@lahwaacz This will take a while, the last release was May 18, 2023 and normally there are 1 ... 3 releases a year. To the best of my knowledge there is no release scheduled yet.

**Comment by lahwaacz** [comment]
Hmm, that's a bummer. Given the amount of issues labeled [fixed but not released](https://github.com/doxygen/doxygen/labels/fixed%20but%20not%20released), I thought a bugfix release might happen soon...

**Comment by albert-github** [comment]
@lahwaacz You can always use the executables as build by means of the github actions, see e.g. https://github.com/doxygen/doxygen/actions/runs/5235740905(bottom of the page the artifacts, it are just the executables there not full fledged installations)

**Comment by lahwaacz** [comment]
I know, I could also build from the git master branch myself, but that still does not fix workflows that need a proper release (e.g. installing from repositories of Linux distributions, container images for CI, etc). Since you make releases, that workflow should be usable as well, otherwise it wouldn't make sense to bother with releases.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
